### ISSUE TITLE
Wrong mapping for cmd, alt, shift, ctrl keys on macOS?

### ISSUE THREAD

**Issue by nicolas-p** [factor/factor#1748]
I already saw that quite a long time ago but I wasn't too sure at the time.

It seems that on my Mac running Sierra (but already in El Capitan), the CMD key is mapped to A+ in Factor (which is supposed to be ALT).

```
environment-gadget "general" f {
    { T{ key-down f { A+ } "s" } save-skov-image }
} define-command-map
```
With the code, ALT+S does nothing but CMD+S triggers the action.

I know that CTRL is correctly mapped to C+ but CMD, ALT and SHIFT might be all wrong.

**Comment by mrjbq7** [comment]
I think ``ALT`` (using ``fn+OPT``) generates a ``M+`` modifier.

**Comment by nicolas-p** [comment]
But the documentations says:

> **M+**
> Meta key modifier. This is the Command key on Mac OS X

**Comment by mrjbq7** [comment]
Hmm...

**Comment by mrjbq7** [comment]
Are you suggesting fixing the documentation or the code?

**Comment by nicolas-p** [comment]
The docs seem fine, it's logical that:
M+ = Meta
A+ = Alt
S+ = Shift
C+ = Ctrl

If the code says M+ = Alt and A+ = Meta, that's super strange!

**Comment by mrjbq7** [comment]
Well, I mean that the documentation about Mac does not match the code.  These are the mappings on macOS:

* S+ = Shift
* C+ = Control
* A+ = Command
* M+ = Alt

https://github.com/factor/factor/blob/master/basis/ui/backend/cocoa/views/views.factor#L36

I think it was probably done because ALT is annoying to type, and CMD is used like ALT on other platforms.

**Comment by nicolas-p** [comment]
Was this done so that keyboard shortcuts automatically respect platform-specific conventions? In this case I would expect Control on Windows = Command on Mac.

**Comment by mrjbq7** [comment]
It was definitely done to help users make cross-platform applications.  For example, in the Listener we do a single binding for ``A+L`` to create a new listener window which is ``Cmd-L`` on the Mac (convenient) and ``Alt-L`` on other platforms.

Any suggestions at making it easier, simpler, more explicit, etc. are welcome.

**Comment by nicolas-p** [comment]
Then I would have intuitively chosen:
<table>
<tr><th>Mac</th><th>Windows</th></tr>
<tr><td>Command</td><td>Control</td></tr>
<tr><td>Control</td><td>Windows</td></tr>
<tr><td>Alt</td><td>Alt</td></tr>
<tr><td>Shift</td><td>Shift</td></tr>
</table>
But at least if the code stays the way it is, the documentation should be corrected.

**Comment by mrjbq7** [comment]
Yeah, the problem is that Alt on Windows is typically used as a menu accelerator key (which is what Command is often used for on Mac).  It's messy.

**Comment by nicolas-p** [comment]
Yes, there's probably no good way to do it.
